lib/game_teams_collection.rb

```diff
@@ -13,21 +13,66 @@ def initialize(file_path)
   def create_game_teams_array(file_path)
     load_from_csv(file_path, GameTeams)
   end
-  # require "pry"; binding.pry
-end
-
 
+  def game_teams_hash
+    @game_teams_array.reduce({}) do |hash, game_teams|
+      hash[game_teams.team_id] << game_teams if hash[game_teams.team_id]
+      hash[game_teams.team_id] = [game_teams] if hash[game_teams.team_id].nil?
+      hash
+    end
+  end
 
-  def game_teams_by_id
+  def percentage(numerator, denominator) #to-do: make Calculatable module
+    return ((numerator.to_f / denominator) * 100).round(2)
+  end
 
-    hash.keys.reduce({}) do |new_hash, key|
-      new_hash[key] = hash[key].find_home_games
+  def team_win_percentage(games_hash, team, hoa = nil)
+    team_games = Hash.new
+    if hoa
+      team_games[team] = games_hash[team].find_all { |game_team| game_team.hoa == hoa }
+    else
+      team_games[team] = games_hash[team]
     end
+    wins = team_games[team].count { |game_team| game_team.result == "WIN" }
+    percentage(wins, team_games[team].length)
+  end
 
+  def winningest_team_id
+    hash = game_teams_hash
+    win_percentages = Hash.new
+    hash.each do |key, value|
+      win_percentages[key] = team_win_percentage(hash, key)
+    end
+    win_percentages.max_by { |key, value| value }[0]
   end
 
-  def find_home_games
-    game_teams_lists_by_id[team_id].find_all do |game_teams|
-      game_teams.hoa == "home"
+  def hoa_differences(hash)
```

Comment:
could be more explicit in terms of where hash argument is coming from

Reply:
renamed to all_games